src/main/java/games/strategy/triplea/ui/ResourceBar.java

```diff
@@ -64,15 +64,17 @@ public void gameDataChanged(final Change change) {
     try {
       final PlayerID player = gameData.getSequence().getStep().getPlayerId();
       if (player != null) {
-        for (int i = 0; i < resources.size(); i++) {
-          final String quantity = resources.get(i).getFormatter().format(resources.get(i).getValue(player, gameData));
+        for (int i = 0; i < resourceStats.size(); i++) {
+          final IStat resourceStat = resourceStats.get(i);
+          final Resource resource = gameData.getResourceList().getResource(resourceStat.getName());
```

Comment:
One simplification here may be to use `ResourceStat#resource` rather than having to do the resource lookup again:

```diff
diff --git a/src/main/java/games/strategy/triplea/ui/ResourceBar.java b/src/main/java/games/strategy/triplea/ui/ResourceBar.java
index 38dcf4535..32ad37a92 100644
--- a/src/main/java/games/strategy/triplea/ui/ResourceBar.java
+++ b/src/main/java/games/strategy/triplea/ui/ResourceBar.java
@@ -12,7 +12,6 @@ import games.strategy.engine.data.GameData;
 import games.strategy.engine.data.PlayerID;
 import games.strategy.engine.data.Resource;
 import games.strategy.engine.data.events.GameDataChangeListener;
-import games.strategy.engine.stats.IStat;
 import games.strategy.triplea.Constants;
 
 /**
@@ -22,7 +21,7 @@ public class ResourceBar extends AbstractStatPanel implements GameDataChangeList
   private static final long serialVersionUID = -7713792841831042952L;
 
   private final UiContext uiContext;
-  private final List<IStat> resourceStats = new ArrayList<>();
+  private final List<ResourceStat> resourceStats = new ArrayList<>();
   private final List<JLabel> labels = new ArrayList<>();
 
   public ResourceBar(final GameData data, final UiContext uiContext) {
@@ -65,8 +64,8 @@ public class ResourceBar extends AbstractStatPanel implements GameDataChangeList
       final PlayerID player = gameData.getSequence().getStep().getPlayerId();
       if (player != null) {
         for (int i = 0; i < resourceStats.size(); i++) {
-          final IStat resourceStat = resourceStats.get(i);
-          final Resource resource = gameData.getResourceList().getResource(resourceStat.getName());
+          final ResourceStat resourceStat = resourceStats.get(i);
+          final Resource resource = resourceStat.resource;
           final String quantity = resourceStat.getFormatter().format(resourceStat.getValue(player, gameData));
           labels.get(i).setVisible(resource.isDisplayedFor(player));
           try {
```

Reply:
Updated.